module/zfs/zfs_sa.c

```diff
@@ -309,6 +309,7 @@ zfs_sa_upgrade(sa_handle_t *hdl, dmu_tx_t *tx)
 
 	/* First do a bulk query of the attributes that aren't cached */
 	bulk = kmem_alloc(sizeof (sa_bulk_attr_t) * 20, KM_SLEEP);
```

Comment:
Not a defect in this patch (or necessarily in the existing code), but it's interesting that bulk is already oversized here.